### PR TITLE
Allow to configure consensus

### LINKED ISSUE #1067: Configurable consensus - CLI flags
For testing and measurements we might want to play with some of the parameters of the network:
1. K cluster size.
2. Max parents limit.
3. Finality depth.
4. MergeSet size limit.
5. Mass limit.
6. Block size.
7. Max payload size.
8. Disable mempool standardness rules (required for OP_RETURN performances)

Obviously these are dangerous parameters to play with, so we probably want to take some caution:
1. Maybe not print them in the `help` menu.
2. Disable them unless a `--dangerous` flag is passed.

We also want to implement a bunch of asserts on initialization that check that these parameters are upholding all the necessary invariants (e.g. K < Finality duration)

### PR DESCRIPTION
**Pull request by someone235**
closes #1067

### CODE REVIEW

**Comment by elichai** [comment]
Can you give an example of how to configure everything?
Thanks!

**Comment by someone235** [comment]
> Can you give an example of how to configure everything?
> Thanks!

`kaspad --devnet --override-dag-params-file=/tmp/x.json`

and this is an example JSON file

`{"targetTimePerBlockInMilliSeconds":1,"powMax":"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"}`

The full JSON definition can be found [here](https://github.com/kaspanet/kaspad/pull/1175/files#diff-ea73f239d8f12ff284184f15ac2f691b597a74cede1db17c8ebbb8fdb7c162b3R26)

File: domain/dagconfig/params.go [hunk 1]
```diff
@@ -45,6 +45,42 @@ var (
 	devnetPowMax = new(big.Int).Sub(new(big.Int).Lsh(bigOne, 239), bigOne)
 )
 
+const (
+
+	// MaxCoinbasePayloadLength is the maximum length in bites allowed for a block's coinbase's payload
+	maxCoinbasePayloadLength = 150
```
Comment (svarogg) [inline domain/dagconfig/params.go:51]: I think we still want these in constants, and use the default Params out of constants

**Review by svarogg — CHANGES_REQUESTED** [review]

File: app/protocol/flows/blockrelay/handle_relay_invs.go [hunk 2]
```diff
@@ -225,29 +224,6 @@ func (flow *handleRelayInvsFlow) processAndRelayBlock(requestQueue *hashesQueueS
 
 		missingParentsError := &ruleerrors.ErrMissingParents{}
 		if errors.As(err, missingParentsError) {
-			blueScore, err := blocks.ExtractBlueScore(block)
```
[the remark this answers is not in the document]
Comment (someone235) [reply app/protocol/flows/blockrelay/handle_relay_invs.go:228]: It's gonna be changed anyway by Stas's code

**Review by svarogg — APPROVED** [review]

**Pull request merged**